### DISCUSSION TITLE
FR: Add more options to the e-mail field.

### DISCUSSION THREAD

**Discussion by TomDeSmet** [sebastian-lenz/craft-linkfield#195] [Ideas]
This is an edge case but we just got a request from a client for this and it may be an easy feature to implement (that I don't know of course).

Take a look here: https://css-tricks.com/snippets/html/mailto-links/
It would be great if we had a subject field and maybe CC, BCC when choosing the e-mail option (even a body field, but that's maybe to much a hassle for the UI).

**Update**
As a workaround I now did this in Twig:
```
{% set isMailto = 'mailto:' in cta.linkIt.getUrl() %}
    {% set linkUrl = cta.linkIt.getUrl() %}
    {% if isMailto %}
        {% set linkUrl = cta.linkIt.getUrl() ~ '?subject=' ~ 'Your subject here'|replace(' ', '%20') %}
    {% endif %}
    <a href="{{ linkUrl }}"
         target="{{ cta.linkIt.getTarget() }}"
         {{ cta.linkIt.getLinkAttributes() }}
         class="btn btn--lg mt-4">{{ cta.linkIt.getCustomText() }}</a>
```

**Comment by sebastian-lenz** [comment]
Okay, I think this is a very specific requirement and I don't think it really fits the general purpose approach of this plugin. But while looking at your problem I've noticed that the link field currently actually does not offer any tools to modify the returned url, which basically is what you need in this case.

Starting with the v2.0.0-beta.10 the link field now accepts an option parameter that can be passed to the url getter that can manipulate the returned url. In a twig template we can now do this to append a query string to the url:

```html
myLinkField.getUrl({
  query: 'subject=Custom+mail+subject&body=Custom+mail+body'
})
```

The query parameter also accepts an array:
```html
myLinkField.getUrl({
  query: {
    subject: 'Custom mail subject',
    body: 'Custom mail body',
  }
})
```

Actually all url parts that are returned by [`parse_url`](https://www.php.net/manual/de/function.parse-url.php) can be manipulated:
```html
myLinkField.getUrl({
  fragment: 'myAnchor',
  scheme: 'https',
})
```

When rendering a complete link via `Link::getLink` or retrieving the link attributes via `Link::getLinkAttributes` the same behavior can be triggered by passing an array to the `href` option:
```html
myLinkField.getLink({
  class: 'customLinkClass',
  href: {
    query: {
      subject: 'Custom mail subject',
      body: 'Custom mail body',
    }
  }
})
```

**Comment by zizther** [comment]
I think it would be great to support a subject and possibly other fields for an email basis, there are options for different types already, and it would be great to have this functionality built in with the conditional options showing rather than having to build in extra logic.

My use-case if building out link in a matrix field where I don't want to have to create additional fields which may not be relevant to other types.

**Comment by sander-lameco** [comment]
I found out using the `Disable Mail input validation` works just fine, that way the ?subject and other parameters can be set in the CMS.

**Comment by CreateSean** [comment]
Wanted to add that I'm also looking for a solution to this. Will test above but having an additional **subject** field available for email would be really great.
